Add new builds Tasks dotnet core

mikaelkrief opened the pull request:
In asp net build CI, I updated the cmd tasks (for restore, build and publish commands) with the new DotNet Core build tasks.

![image](https://cloud.githubusercontent.com/assets/2725302/20572041/158cc6d0-b1aa-11e6-8c33-64d2aa400101.png)

mikaelkrief committed:
update build template with new task dotnetcore

mikaelkrief committed:
copy parameters and json files to publish folder

DarqueWarrior commented on generators/vsts/templates/asp_build.json:
```diff
@@ -28,18 +28,33 @@
          },
          "inputs": {
-            "filename": "dotnet",
-            "arguments": "restore --no-cache",
```
The --no-cache argument is important for hosted builds https://docs.microsoft.com/en-us/dotnet/articles/core/tools/dotnet-restore and should be still be passed as an argument.

DarqueWarrior reviewed (commented):
You don't need the build task.  Test and publish will build if needed.

mikaelkrief committed:
-delete build task
-add no-cache argument for restore
-add option section on build json

mikaelkrief commented:
@DarqueWarrior : Pull Request updated with:

-delete build task
-add no-cache argument for restore
-add build 'option' section on build json

Commit:
Last coat of polish
With the new publish tasks we no longer need the parameters.xml so I removed it from the asp project and updated the copy files task.

DarqueWarrior committed:
Updating version so I can publish to npm.
